### ISSUE TITLE
Should we start a Polaris config that folks can opt in to when using gjs/gts?

### ISSUE THREAD

**Issue by NullVoxPopuli** [ember-template-lint/ember-template-lint#2830]
I think we need to change defaults when using gjs/gts, for example

several rules throw false negatives because templates can reference things imported in JS, but not defined in the template. for example:
```js
import { overInvalidatingClock, clock } from './-resources';

const SomeClocks = <template>
  {{clock}}
  {{overInvalidatingClock}}
</template>;
```
this throws errors for the following rules:
 - `no-curly-component-invocation`
 - `no-implicit-this`
 

the behavior of checking if locals are defined or not is handled be eslint / eslint-plugin-ember via transform.

So, for ember-template-lint and a `polaris` config, I think we should start disabling some rules specifically for gjs/gts files -- and we could use this for the polaris config:

```js
'use strict';

module.exports = {
  extends: 'recommended',
  overrides: [
    {
      files: ['**/*.gjs', '**/*.gts'],
      rules: {
        'no-curly-component-invocation': 'off',
        'no-implicit-this': 'off',
      },
    },
  ],
};
```

Related: https://github.com/ember-template-lint/ember-template-lint/issues/2786

**Comment by bmish** [comment]
Do we need a `polaris` config? Why not just make those overrides part of the `recommended` config so everyone gets them atuomatically?

**Comment by NullVoxPopuli** [comment]
yeah, I suppose that makes sense -- this sort of thing can only reduce bugs people face atm: https://github.com/ember-template-lint/ember-template-lint/pull/2844

**Comment by bmish** [comment]
Just to confirm, it doesn't make any sense to use these rules for these file types, right? If that's right, I think it makes sense to treat this as a bug fix.

And are there likely to be other rules that don't make sense for these file types?

**Comment by NullVoxPopuli** [comment]
>  If that's right, I think it makes sense to treat this as a bug fix.

correct, and there is a corresponding eslint-plugin-ember PR for resolving the general collective question of "which locals are not in scope?" (already merged):
 - https://github.com/ember-cli/eslint-plugin-ember/pull/1801
 - https://github.com/ember-cli/eslint-plugin-ember/pull/1795
 
 > And are there likely to be other rules that don't make sense for these file types?
 
 potentially! We'll just need to keep iterating I suppose and get more testers
